code/game/gamemodes/miniantags/revenant/revenant_abilities.dm

```diff
@@ -259,30 +262,142 @@
+	action_icon_state = "haunt"
+	base_cooldown = 60 SECONDS
+	unlock_amount = 150
+	cast_amount = 50
+	stun = 3 SECONDS
+	reveal = 10 SECONDS
+	aoe_range = 7
+	/// The maximum number of objects to haunt
+	var/max_targets = 7
+	/// A list of all attack timers started by this spell being cast
+	var/list/attack_timers = list()
+
+/obj/effect/proc_holder/spell/aoe/revenant/haunt_object/create_new_targeting()
+	var/datum/spell_targeting/aoe/obj/targeting = new()
+	targeting.range = aoe_range
+	return targeting
+
+/obj/effect/proc_holder/spell/aoe/revenant/haunt_object/cast(list/targets, mob/living/simple_animal/revenant/user = usr)
+	if(!attempt_cast(user))
+		return
+	var/successes = 0
+	for(var/obj/item/nearby_item in targets)
+		// Don't throw around anchored things or dense things
+		// (Or things not on a turf but I am not sure if range can catch that)
+		if(nearby_item.anchored || nearby_item.density || !isturf(nearby_item.loc))
+			continue
+		// Don't throw abstract things
+		if(nearby_item.flags & ABSTRACT)
+			continue
+		// Don't throw things we can't see
+		if(nearby_item.invisibility >= INVISIBILITY_REVENANT)
+			continue
+
+		var/distance_from_user = clamp(get_dist(get_turf(nearby_item), get_turf(user)), 1, 127) // get_dist() never returns more than 127, but same tile dists return -1
+		var/chance_of_haunting = 150 * (1 / distance_from_user) // The further away things are, the less likely they are to be picked
+		if(!prob(chance_of_haunting))
+			continue
+
+		if(successes >= max_targets) // End loop if we've already got 7 spooky items
+			return
+
+		make_spooky(nearby_item, user)
+		successes++
+
+/// Handles making an object haunted and setting it up to attack
+/obj/effect/proc_holder/spell/aoe/revenant/haunt_object/proc/make_spooky(obj/item/item_to_possess, mob/living/simple_animal/revenant/user)
+	new /obj/effect/temp_visual/revenant(get_turf(item_to_possess)) // Thematic spooky visuals
+	var/mob/living/simple_animal/possessed_object/possessed_object = new(item_to_possess) // Begin haunting object
+	possessed_object.add_filter("haunt_glow", 2, list("type" = "outline", "color" = "#7A4FA9", "size" = 1)) // Give it spooky purple outline
+	possessed_object.throwforce = clamp(item_to_possess.throwforce, item_to_possess.throwforce + 5, 15) // Damage it should do? throwforce+3 or 15, whichever is lower
+	possessed_object.maxHealth = 100 // Double the regular HP of possessed objects
+	possessed_object.health = 100
+	possessed_object.escape_chance = 100 // We cannot be contained
+
+	addtimer(CALLBACK(src, .proc/attack, possessed_object, user), 2 SECONDS, TIMER_UNIQUE) // Short warm up for floaty ambience
+	attack_timers.Add(addtimer(CALLBACK(src, .proc/attack, possessed_object, user), 5 SECONDS, TIMER_UNIQUE|TIMER_LOOP|TIMER_STOPPABLE)) // 5 second looping attacks
+	addtimer(CALLBACK(src, .proc/stop_timers), 65 SECONDS, TIMER_UNIQUE) // Stop the looping attacks after 65 seconds, roughly 14 attack cycles depending on lag
```

Comment:
I'd rather you call this one after the `cast` itself if there were any successes.